featuretools/entityset/entity.py

```diff
@@ -278,6 +279,15 @@ def _create_variables(self, variable_types, index, time_index, secondary_time_in
         """
         variables = []
         variable_types = variable_types.copy() or {}
+        string_to_class_map = find_variable_types()
+        for vid in variable_types.copy():
+            vtype = variable_types[vid]
+            if isinstance(vtype, str):
+                if vtype in string_to_class_map:
+                    variable_types[vid] = string_to_class_map[vtype]
+                else:
+                    variable_types[vid] = string_to_class_map['unknown']
```

Comment:
Let's add a warning here so the user knows the string wasn't recognized and `Unknown` was used instead